test/e2e/logs_mtls_cert_key_pair_dont_match_test.go

```diff
@@ -0,0 +1,79 @@
+//go:build e2e
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/telemetry-manager/internal/conditions"
+	"github.com/kyma-project/telemetry-manager/internal/testutils"
+	kitk8s "github.com/kyma-project/telemetry-manager/test/testkit/k8s"
+	"github.com/kyma-project/telemetry-manager/test/testkit/mocks/backend"
+	"github.com/kyma-project/telemetry-manager/test/testkit/mocks/loggen"
+	"github.com/kyma-project/telemetry-manager/test/testkit/suite"
+	"github.com/kyma-project/telemetry-manager/test/testkit/verifiers"
+)
+
+var _ = Describe(suite.ID(), Label(suite.LabelLogs), Ordered, func() {
+	var (
+		mockNs       = suite.ID()
+		pipelineName = suite.ID()
+	)
+
+	makeResources := func() []client.Object {
+		var objs []client.Object
+		objs = append(objs, kitk8s.NewNamespace(mockNs).K8sObject())
+
+		serverCertsDefault, clientCertsDefault, err := testutils.NewCertBuilder(backend.DefaultName, mockNs).Build()
+		Expect(err).ToNot(HaveOccurred())
+
+		_, clientCertsFoo, err := testutils.NewCertBuilder("foo", mockNs).WithCommonName("foo.com").Build()
+		Expect(err).ToNot(HaveOccurred())
+
+		backend := backend.New(mockNs, backend.SignalTypeLogs, backend.WithTLS(*serverCertsDefault))
+		objs = append(objs, backend.K8sObjects()...)
+
+		invalidClientCerts := testutils.BuildInvalidKeyPair(clientCertsDefault.CaCertPem, clientCertsDefault.ClientCertPem, clientCertsFoo.ClientKeyPem)
```

Comment:
Let's maybe generate 2 key/cert pairs for default.com. In this test you are mixing the wrong common name and non-matching key/cert pair.